### PR TITLE
Fix DB migration by pulling out everything from the DB first

### PR DESCRIPTION
**Pull request by S7evinK**
Given the discussed issue from yesterday, this is merely a workaround for the migration, as I couldn't find a viable way to do this in SQL directly.

### CODE REVIEW

File: state/migrations/20230802121023_device_data_jsonb.go [hunk 1]
```diff
@@ -33,10 +34,32 @@ func upJSONB(ctx context.Context, tx *sql.Tx) error {
 	if err != nil {
 		return err
 	}
-	_, err = tx.ExecContext(ctx, "UPDATE syncv3_device_data SET dataj = encode(data, 'escape')::JSONB;")
+
+	rows, err := tx.Query("SELECT user_id, device_id, data FROM syncv3_device_data")
```
Comment (kegsay) [inline state/migrations/20230802121023_device_data_jsonb.go:38]: Missing defer rows close

File: state/migrations/20230802121023_device_data_jsonb.go [hunk 2]
```diff
@@ -33,10 +34,32 @@ func upJSONB(ctx context.Context, tx *sql.Tx) error {
...
+		if err = rows.Scan(&deviceData.UserID, &deviceData.DeviceID, &data); err != nil {
+			return err
+		}
+		deviceDatas[deviceData] = data
+	}
+
+	for dd, d := range deviceDatas {
+		_, err = tx.ExecContext(ctx, "UPDATE syncv3_device_data SET dataj = $1 WHERE user_id = $2 AND device_id = $3;", d, dd.UserID, dd.DeviceID)
+		if err != nil {
+			return err
```
Comment (kegsay) [inline state/migrations/20230802121023_device_data_jsonb.go:59]: How does this help when there are problems casting to jsonb?

File: state/migrations/20230802121023_device_data_jsonb_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	tx, err := db.Begin()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tx.Commit()
+
+	// insert some "invalid" data
+	dd := internal.DeviceData{
+		DeviceLists: internal.DeviceLists{
+			New:  map[string]int{"@💣:localhost": 1},
```
Comment (kegsay) [inline state/migrations/20230802121023_device_data_jsonb_test.go:59]: It's not the unicode that's the problem on its own I think. It's the escaping part in postgres that it fails on. Can you check that this string does fail when escaped?

**Review by kegsay — CHANGES_REQUESTED** [review]

File: state/migrations/20230802121023_device_data_jsonb.go (hunk 2 above)
Comment (S7evinK) [reply state/migrations/20230802121023_device_data_jsonb.go:59]: It's not the casting to jsonb, it's the `encode(data, 'escape')` which adds e.g. `\3` which then can't be casted to jsonb

File: state/migrations/20230802121023_device_data_jsonb_test.go (hunk 3 above)
Comment (S7evinK) [reply state/migrations/20230802121023_device_data_jsonb_test.go:59]: We're not using any escaping now, it is just an issue with the migration itself

**Pull request merged**